cpp/src/utils/metrics/dimension.hh

```diff
@@ -0,0 +1,19 @@
+#ifndef TS_CPP_UTILS_METRICS_DIMENSION_HH_
+#define TS_CPP_UTILS_METRICS_DIMENSION_HH_
+
+#include <string>
+
+namespace torchserve {
+    class Dimension {
+        public:
+        Dimension(const std::string& name, const std::string& value);
+        const std::string GetName() const;
+        const std::string GetValue() const;
+
+        private:
+        const std::string name;
+        const std::string value;
+    };
```

Comment:
please address my[ previous comments](https://github.com/pytorch/serve/pull/1874#discussion_r983866145).

Reply:
I do agree with your comment that a `Dimension` has a name and can take on multiple values. The intention of defining this class was to use it in the context of a particular metric instance. For example, if we have a metric that tracks the inference latency for a fastrcnn model with batch size 2, then the dimensions for that specific metric instance could be something like `{"model_name": "fastrcnn", "batch_size": "2"}`. Example usage in [testcase](https://github.com/pytorch/serve/pull/1874/files#diff-f399ff4bf1dde7c7aa221ae2e2441f29aa62b6c1f41f3938f5afb7744350ead7R43). Also, I believe that we will not know all the possible dimension values during initialization time, since we don't enforce that in the metrics configuration. Please let me know your thoughts on this.

Reply:
It is impossible to know all the possible values of a dimension. As an API, the data structure should be able to reflect this use case, eg. vector.

Reply:
That's true, for storing dimension names and values within a `Metric` object, the `Dimension` object may not be absolutely necessary. I was also considering its use in the context of supporting the `MetricCache` API, i.e `Metric addMetric(metricName, unit, dimensions, metricType);` and `Metric getMetric(metricName, dimensions);`. In both of these cases, the `dimensions` argument will need to be a vector of `<dimension_name:single_dimension_value>` tuples. So here, it may be clear to the user, if we express the `dimensions` input argument to be of type `std::vector<Dimension>`. What do you think?